ext/PreallocationToolsReverseDiffExt.jl

```diff
@@ -0,0 +1,17 @@
+module PreallocationToolsReverseDiffExt
+
+using PreallocationTools
+import ReverseDiff
+
+# PreallocationTools https://github.com/SciML/PreallocationTools.jl/issues/39
+function Base.getindex(b::PreallocationTools.LazyBufferCache, u::ReverseDiff.TrackedArray)
+    s = b.sizemap(size(u)) # required buffer size
+    T = ReverseDiff.TrackedArray
+    buf = get!(b.bufs, (T, s)) do
+        # declare type since b.bufs dictionary is untyped
```

Comment:
Where is the type being declared?

Reply:
what do you mean?

Reply:
I interpreted the comment as meaning `b.bufs` is an untyped dictionary, so that `buf` is `Any`.
And that the types were supposed to be used somewhere to add more information.